src/frontends/tensorflow/src/op/parallel_dynamic_stitch.cpp

```diff
@@ -0,0 +1,83 @@
+namespace op {
+
+OutputVector translate_parallel_dynamic_stitch_op(const NodeContext& node) {
+    // format for inputs: [indices1, indices2, ..., indicesN, data1, data2, ..., dataN]
+    // so we expect at least 2 input and the total number of inputs must be divisible by 2
+    default_op_checks(node, 2, {"ParallelDynamicStitch", "DynamicStitch"});
+    auto in_size = node.get_input_size();
+    TENSORFLOW_OP_VALIDATION(node,
+                             in_size % 2 == 0,
+                             "The total number of inputs to DynamicStitch operation "
+                             "must be divisible by 2.");
+
+    int N = static_cast<int>(in_size / 2);
+    OutputVector indices_to_concat;
+    OutputVector data_to_concat;
+    auto data_element_type = node.get_input(N).get_element_type();
+    auto const_minus_one = std::make_shared<Constant>(ov::element::i32, Shape{1}, -1);
+    auto const_zero = std::make_shared<Constant>(ov::element::i32, Shape{1}, 0);
+    auto const_one = std::make_shared<Constant>(ov::element::i32, Shape{1}, 1);
+    for (int i = 0; i < N; ++i) {
+        auto indices = node.get_input(i);
+        auto data = node.get_input(N + i);
+
+        const auto& indices_pshape = indices.get_partial_shape();
+        auto rank = indices_pshape.rank();
+        TENSORFLOW_OP_VALIDATION(node,
+                                 indices_pshape.rank().is_static(),
+                                 "Only static rank for `indices` input is supported.");
+        auto rank_val = rank.get_length();
+        auto norm_indices = std::make_shared<Reshape>(indices, const_minus_one, false);
+        if (rank_val < 1) {
+            data = std::make_shared<Unsqueeze>(data, const_zero);
+        } else if (rank_val > 1) {
+            auto data_shape = std::make_shared<ShapeOf>(data, ov::element::i32);
+            auto start = std::make_shared<Constant>(ov::element::i32, Shape{1}, rank_val);
+            auto stop = std::make_shared<Constant>(ov::element::i32, Shape{1}, std::numeric_limits<int>::max());
+            auto shape_of_single_element = std::make_shared<Slice>(data_shape, start, stop, const_one);
+            auto new_shape = std::make_shared<Concat>(OutputVector{const_minus_one, shape_of_single_element}, 0);
+            data = std::make_shared<Reshape>(data, new_shape, false);
+        }
+        data_to_concat.push_back(data);
+        indices_to_concat.push_back(norm_indices);
+    }
+    auto update = std::make_shared<Concat>(data_to_concat, 0);
+    auto indices = std::make_shared<Concat>(indices_to_concat, 0);
+    auto data_shape = std::make_shared<ShapeOf>(update, ov::element::i32);
+
+    auto zero = std::make_shared<Constant>(data_element_type, Shape{1}, 0);
+    auto zeros = std::make_shared<Broadcast>(zero, data_shape);
+
+    auto scatter = std::make_shared<ScatterUpdate>(zeros, indices, update, const_zero);
+
+    // we should cut zeros at the end of the result output to align it with tf framework
```

Comment:
can you elaborate more in what cases the padding remained?

Reply:
no such cases
I will add additional details in the comment

Reply:
if no such cases, why do we need it to overload the functionality?

Reply:
Slice has been moved before ScatterUpdate